web/META-INF/resources/js/controllers.js

```diff
@@ -0,0 +1,261 @@
+        $scope.options = {
+            opt: 'Or'
+        };
+
+        $scope.complete = function (string) {
+            $scope.hidethis = false;
+            Autocomplete.get({term: string},
+                function success(response) {
+                    $scope.filterSearch = response;
+
+                },
+                function error(errorResponse) {
+                    console.log("Error:" + JSON.stringify(errorResponse));
+                }
+            );
+        };
+
+        $scope.fillTextbox = function (string) {
+            $scope.options.query = string;
+            $scope.hidethis = true;
+        };
+
+        $scope.search = function (options) {
+            $rootScope.searchOptions = angular.copy(options);
+            $location.path("/metadataView/" + options.query + '/' + options.opt);
+        };
+
+        $scope.$watch(function () {
+            return SearchOptions.getSearchOptions();
+        }, function (newValue, oldValue) {
+            if (newValue !== null) {
+                $scope.options.query = newValue.query;
+                $scope.options.opt = newValue.opt;
+            }
+        }, true);
+    }]);
+
+mudrodControllers.controller('vocabularyCtrl', ['$scope', '$rootScope', 'VocabList',
+    function vocabularyCtrl($scope, $rootScope, VocabList) {
+
+        var word = $rootScope.searchOptions.query;
+        var opt = $rootScope.searchOptions.opt;
+        VocabList.get({query: word},
+            function success(response) {
+                $scope.ontologyList = response.graph.ontology;
+
+            },
+            function error(errorResponse) {
+                console.log("Error:" + JSON.stringify(errorResponse));
+            }
+        );
+    }]);
+
+mudrodControllers.controller('metadataViewCtrl', ['$rootScope', '$scope', '$routeParams', 'MetaData', 'PagerService', 'SearchOptions',
+    function metadataViewCtrl($rootScope, $scope, $routeParams, MetaData, PagerService, SearchOptions) {
+        var vm = this;
+        vm.PDItems = [];
+        vm.pager = {};
+        vm.setPage = setPage;
+        vm.totalMatches = 0;
+
+        var word = String();
+        var opt = String();
+
+        if (!$routeParams.query) {
+            word = $rootScope.searchOptions.query;
+            opt = $rootScope.searchOptions.opt;
+            SearchOptions.setSearchOptions({'query': word, 'opt': opt});
+        } else {
+            word = $routeParams.query;
+            opt = $routeParams.opt;
+
+            var searchKeyWords = String();
+
+            if (word.search(',') !== -1) {
+                var topics = word.split(',');
+                for (var i = 0; i < topics.length; i++) {
+                    searchKeyWords += topics[i];
+                }
+            } else {
+                searchKeyWords = word;
+            }
+
+            if (searchKeyWords.search('/') !== -1) {
+                var topics = searchKeyWords.split('/');
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "topics" as this name is already used in declaration at line 97. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)